src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs

```diff
@@ -72,5 +72,27 @@ public class RetryHelper
             logger.LogInformation("Failed to connect, retry limit exceeded.");
             throw new OperationCanceledException("Failed to connect, retry limit exceeded.");
         }
+
+        public static void RetryOperation(
+            Action retryBlock,
+            Action<Exception> exceptionBlock,
+            int retryCount = 3,
+            int retryDelayMilliseconds = 0)
+        {
+            for (var retry = 0; retry < retryCount; ++retry)
+            {
+                try
+                {
+                    retryBlock();
+                    break;
+                }
+                catch (Exception exception)
```

Comment:
I'm talking about something like this: https://github.com/aspnet/Logging/blob/e9e62d27d3719fe0abbb94ca93f5c2bc2d80fffd/test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/RetryHelper.cs